torchgeo/datasets/wms.py

```diff
@@ -0,0 +1,145 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# Author: Ian Turton, Glasgow University ian.turton@gla.ac.uk
+
+"""A simple class to fetch WMS images."""
+from io import BytesIO
+from typing import Any, Callable, Optional, Union
+
+import torchvision.transforms as transforms
+from owslib.map.wms111 import ContentMetadata
+from owslib.wms import WebMapService
```

Comment:
Imports should be done lazily inside of the class so that the rest of TorchGeo still works even if owslib isn't installed. If you grep for any of our other optional dataset dependencies, you should see example code for this you can copy-n-paste.

Reply:
> @ianturton you can check `torchgeo.datasets.chabud` as an example. See how we check if h5py is available in the constructor and then import h5py in the method that uses it.